packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -628,15 +648,23 @@ class _RootBuilder extends dart_ast_visitor.RecursiveAstVisitor<Object?> {
           .firstWhere((e) => e is dart_ast.TypeName) as dart_ast.TypeName;
       argType = typeName.name.name;
       isNullable = typeName.question != null;
+      argTypeArguments = typeAnnotationsToTypeArguments(typeName.typeArguments);
     }
     final bool isAsynchronous = _hasMetadata(node.metadata, 'async');
     if (_currentApi != null) {
       _currentApi!.methods.add(Method(
           name: node.name.name,
-          returnType: node.returnType.toString(),
-          argType: argType,
-          isReturnNullable: node.returnType!.question != null,
-          isArgNullable: isNullable,
+          returnType: Field(
+              name: '',
```

Comment:
I just noticed that this already introduces the same confusing use of `Field` that came up in the next PR's review.